AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -814,6 +813,33 @@ - (void)setUsesImplicitHierarchyManagement:(BOOL)value
   _usesImplicitHierarchyManagement = value;
 }
 
+- (BOOL)_hasTransitionInProgress
+{
+    ASDN::MutexLocker l(_propertyLock);
+    return _transitionInProgress;
+}
+
+/// Starts a new transition and returns the transition id
+- (int32_t)_startNewTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  _transitionInProgress = YES;
+  _transitionID = OSAtomicAdd32(1, &_transitionID);
+  return _transitionID;
+}
+
+- (void)_finishOrCancelTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  _transitionInProgress = NO;
```

Comment:
It almost feels to me like we should also increment the transitionID here.

Basically, if there were a call point that _only_ checked the transition ID and not transition in progress, it could accidentally succeed even after a transition is cancelled.

This scenario is unlikely, and does not currently occur so it is not a blocker.  Just something to consider for additional robustness.

I do understand we can't just reset the ID to 0, as there is the possibility of multiple calls then arriving at the same ID like "2" and actually passing.